Library/Formula/llvm.rb

```diff
@@ -116,18 +151,74 @@ def install
     # Apple's libstdc++ is too old to build LLVM
     ENV.libcxx if ENV.compiler == :clang
 
-    if build.with?("lldb") && build.without?("clang")
-      raise "Building LLDB needs Clang support library."
+    if build.with? "clang"
+      (buildpath/"tools/clang").install resource("clang")
     end
 
-    if build.with? "clang"
+    if build.with? "extra-tools"
+      if build.with? "clang"
+        (buildpath/"tools/clang/tools/extra").install resource("extra-tools")
+      else
+        odie "clang's extra tools must be built with clang"
+      end
+    end
```

Comment:
You can simplify this block to be shorter and reduce nesting like that:

```ruby
if build.with? "clang-extra-tools"
  raise "Clang's extra tools must be built with Clang" if build.without? "clang"
  (buildpath/"tools/clang/tools/extra").install resource("clang-extra-tools")
end
```

Please apply this idea also to similar blocks below.